src/server/api/controllers/days-to-end-batch.controllers.js

```diff
@@ -0,0 +1,31 @@
+const knex = require('../../config/db');
+
+const Error = require('../lib/utils/http-error');
+
+const moment = require('moment');
+
+const getDayLeftToEndBatch = async (batchId) => {
+  try {
+    const batch = await knex('batches')
+      .select('seeding_date', 'fk_crop_id')
+      .where('id', batchId);
+    if (batch.length === 0) {
+      throw new Error(`incorrect entry with the id of ${batchId}`, 404);
+    }
+
+    const totalDays = await knex('crop_stages')
+      .sum('duration as duration')
+      .where('fk_crops_id', batch[0].fk_crop_id);
```

Comment:
Please update fk_crops_id to fk_crop_id in your knex migration and controller if it's being used because fk_crops_id has been updated to fk_crop_id to make it consistent and to avoid confusion on migrations schema and DB diagram. So please replace fk_crops_id to fk_crop_id if your application is crashing and you want to save time.